heudiconv/cli/run.py

```diff
@@ -0,0 +1,313 @@
+                                 'queue arg')
+    return parser
+
+
+def process_args(args):
+    """Given a structure of arguments from the parser perform computation"""
+
+    # Deal with provided files or templates
+    # pre-process provided list of files and possibly sort into groups/sessions
+    # Group files per each study/sid/session
+
+    lgr.info(INIT_MSG(packname=__packagename__,
+                      version=__version__))
+
+    outdir = op.abspath(args.outdir)
+
+    if args.command:
+        process_extra_commands(outdir, args)
+        return
+
+    #
+    # Load heuristic -- better do it asap to make sure it loads correctly
+    #
+    heuristic = load_heuristic(op.realpath(args.heuristic_file))
+
+    study_sessions = get_study_sessions(args.dicom_dir_template, args.files,
+                                        heuristic, outdir, args.session,
+                                        args.subjs, grouping=args.grouping)
+
+    # extract tarballs, and replace their entries with expanded lists of files
+    # TODO: we might need to sort so sessions are ordered???
+    lgr.info("Need to process %d study sessions", len(study_sessions))
+
+    # processed_studydirs = set()
+
+    for (locator, session, sid), files_or_seqinfo in study_sessions.items():
+
+        if not len(files_or_seqinfo):
+            raise ValueError("nothing to process?")
+        # that is how life is ATM :-/ since we don't do sorting if subj
+        # template is provided
+        if isinstance(files_or_seqinfo, dict):
+            assert(isinstance(list(files_or_seqinfo.keys())[0], SeqInfo))
+            dicoms = None
+            seqinfo = files_or_seqinfo
+        else:
+            dicoms = files_or_seqinfo
+            seqinfo = None
+
+        if locator == 'unknown':
+            lgr.warning("Skipping unknown locator dataset")
+            continue
+
+        if args.queue:
+            if seqinfo and not dicoms:
+                # flatten them all and provide into batching, which again
+                # would group them... heh
+                dicoms = sum(seqinfo.values(), [])
+                raise NotImplementedError(
+                    "we already grouped them so need to add a switch to avoid "
+                    "any grouping, so no outdir prefix doubled etc")
+
+            progname = op.abspath(inspect.getfile(inspect.currentframe()))
+
+            queue_conversion(progname,
+                             args.queue,
+                             study_outdir,
+                             heuristic.filename,
+                             dicoms,
+                             sid,
+                             args.anon_cmd,
+                             args.converter,
+                             args.session,
+                             args.with_prov,
+                             args.bids)
+            continue
+
+        anon_sid = anonymize_sid(sid, args.anon_cmd) if args.anon_cmd else None
```

Comment:
here is where sid is reset to None if no anon_cmd.  I will return it back to `sid` value instead of None

Reply:
@yarikoptic ah yeah I remember this change now -
if we define `anon_sid` as sid further down the pipeline:
https://github.com/nipy/heudiconv/blob/master/heudiconv/cli/run.py#L281

then conversion will use the pre bids cleaned name
https://github.com/nipy/heudiconv/blob/master/heudiconv/convert.py#L71

in retrospect there are better ways to handle it, but currently this will break bids compatibility with sids that contain non-alphanumeric characters